tests/integration/tasks/test_eurostat.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+import pandas as pd
+import logging
+
+from viadot.tasks import eurostat
+
+
+def test_correct_dataset_code_no_params(caplog):
+    """Test for checking if program for correct dataset_code
+    and no parameters is returning non empty DataFrame
+    """
+    task = eurostat.EurostatToDF(dataset_code="ILC_DI04").run()
+    assert isinstance(task, pd.DataFrame)
+    assert not task.empty
+    assert caplog.text == ""
+
+
+def test_wrong_dataset_code_logger(caplog):
+    """Test for checking if program for incorrect dataset_code
+    is logging correct error
+    """
+    task = eurostat.EurostatToDF(dataset_code="ILC_DI04E")
+
+    with pytest.raises(ValueError, match="DataFrame is empty!"):
+        with caplog.at_level(logging.ERROR):
+            task.run()
+    assert (
+        f"Failed to fetch data for ILC_DI04E, please check correctness of dataset code!"
+        in caplog.text
+    )
+
+
+def test_wrong_parameters_codes_logger(caplog):
+    """Test for checking if program for correct dataset_code
+    and correct parameters, but incorrect parameters codes
+    is logging correct error
+    """
+    task = eurostat.EurostatToDF(
+        dataset_code="ILC_DI04",
+        params={"hhtyp": "total1", "indic_il": "non_existing_code"},
+    )
+
+    with pytest.raises(ValueError, match="DataFrame is empty!"):
+        with caplog.at_level(logging.ERROR):
+            task.run()
+    assert (
+        f"Parameters codes: 'total1 | non_existing_code' are not available. Please check your spelling!"
+        in caplog.text
+    )
+    assert (
+        f"You can find everything via link: https://ec.europa.eu/eurostat/databrowser/view/ILC_DI04/default/table?lang=en"
+        in caplog.text
+    )
+
+
+def test_parameter_codes_as_list_loggere(caplog):
+    """Test for checking if program for correct dataset_code,
```

Comment:
Difficult to understand what it means? 

Maybe better to write a question the test function tries to answer by running like this:
> Is the program correctly logging an error when it receives a list of incorrect codes, despite having the correct dataset code and parameters?

Or better description like this one:

> This function is designed to test the accuracy of the error logging feature in a program. Specifically, it tests to ensure that the program is able to correctly identify and log errors when provided with a correct dataset code, correct parameters, but incorrect codes as a list. The function is intended to be used in software development to identify and correct errors in the program's handling of codes.

It is also good to put a period at the end of each sentence and in general to watch out for punctuation.